apps/consulting/posts/improvements-to-bsr-matrix-multiplication-in-pytorch.md

```diff
@@ -0,0 +1,292 @@
+---
+title: "Improvements to BSR Matrix Multiplication in PyTorch"
+published: October 1, 2023
+author: andrew-james
+description: 'Quansight engineers have implemented several new kernels for matrix multiplication
+involving sparse and dense arguments. These implementations, available in PyTorch 2.1,  offer a
+significant performance improvement over those available in previous versions of PyTorch. In
+addition, they have the unique property that they are implemented using `triton`, however they are
+not the product of `torch.compile`.'
+category: [Open Source Software, PyTorch, Triton, GPU]
+---
+<base target="_blank" />
+
+Quansight engineers have implemented several new kernels for matrix multiplication involving sparse
+and dense arguments. These implementations, available in PyTorch 2.1,  offer a significant
+performance improvement over those available in previous versions of PyTorch. In addition, they have
+the unique property that they are implemented using `triton`, however they are not the product of
+`torch.compile`.
```

Comment:
I need to rewrite the tail of this intro. Originally I planned to go through some of the details of how we wired up the triton kernel so that it was dispatched to from the native functions for matmuls. 

Ended up with enough content and didn't feel it was a worthwhile addition, as the major technical hurdle was fighting with torch's init mechanisms to make sure we don't prematurely cause the cuda runtime to be initialized.